libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationOptions.kt

```diff
@@ -0,0 +1,65 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.services.android.navigation.R
+import com.mapbox.services.android.navigation.v5.navigation.notification.NavigationNotification
+
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.NAVIGATION_LOCATION_ENGINE_INTERVAL_LAG
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.ROUNDING_INCREMENT_FIFTY
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.ROUTE_REFRESH_INTERVAL
+
+/**
+ * Immutable and can't be changed after passing into [MapboxNavigation].
+ */
+data class MapboxNavigationOptions(
+    @JvmField val defaultMilestonesEnabled: Boolean = true,
+    @JvmField val enableFasterRouteDetection: Boolean = false,
+    @JvmField val enableAutoIncrementLegIndex: Boolean = true,
+    @JvmField val enableRefreshRoute: Boolean = true,
+    @JvmField val refreshIntervalInMilliseconds: Long = ROUTE_REFRESH_INTERVAL,
+    @JvmField val isFromNavigationUi: Boolean = false,
+    @JvmField val isDebugLoggingEnabled: Boolean = false,
+    @JvmField val navigationNotification: NavigationNotification? = null,
+    @JvmField val roundingIncrement: Int = ROUNDING_INCREMENT_FIFTY,
+    @JvmField val timeFormatType: Int = NavigationTimeFormat.NONE_SPECIFIED,
+    @JvmField val navigationLocationEngineIntervalLagInMilliseconds: Int = NAVIGATION_LOCATION_ENGINE_INTERVAL_LAG,
+    @JvmField val defaultNotificationColorId: Int = R.color.mapboxNotificationBlue
+) {
+    companion object {
+        fun build() = MapboxNavigationOptions()
```

Comment:
Can we add `@JvmStatic`. That way we can refrain from using `Companion` everywhere. 
Makes the code more readable

Reply:
👍